dev/core/test/com/google/gwt/dev/resource/impl/ClassPathEntryTest.java

```diff
@@ -32,18 +32,15 @@
  */
 public class ClassPathEntryTest extends AbstractResourceOrientedTestBase {
 
-  /**
-   * This test will likely not work on Windows since directories that start with . are not
-   * implicitly hidden there. But since Java 6 does not have a File.setHidden() function, fixing
-   * this test for Windows is deferred till GWT officially depends on Java 7.
-   */
   public void testIgnoresHiddenDirectories() throws IOException {
     // Setup a /tmp/.svn/ShouldNotBeFound.java folder structure.
     File tempDir = Files.createTempDir();
     File nestedHiddenDir = new File(tempDir, ".svn");
     nestedHiddenDir.mkdir();
     File javaFile = new File(nestedHiddenDir, "ShouldNotBeFound.java");
     javaFile.createNewFile();
+    // windows needs the hidden attribute to be set on file (not parent directory)
+    java.nio.file.Files.setAttribute(javaFile.toPath(), "dos:hidden", Boolean.TRUE);
```

Comment:
How about using `Files.getFileAttributeView(javaFile.toPath(), DosFileAttributeView.class)` and if it's non-null then call `setHidden(true)`?